src/puppetlabs/trapperkeeper/services.clj

```diff
@@ -22,6 +22,8 @@
   (service-context [this] "Returns the context map for this service")
   (get-service [this service-id] "Returns the service with the given service id")
   (get-services [this] "Returns a sequence containing all of the services in the app")
+  (call-service-fn [this other-service-id fn-name args default] "Calls named function in other service, returning default if that service is not included.")
+  (service-included? [this service-id] "Returns nil if service does not exist and the service if it does")
```

Comment:
I kind of punted since the correct solution is to have get-service actually check if a service *can* be optional and only throw if it cannot. If it can, then we can write code like

`(if-let [svc (get-service this :SomeOptionalService)] ... ...)`

I'm okay changing this name if you have suggestions... get-service-if-included feels wordy, but i seem to be the only person who likes the maybe-* pattern for things that might return nil (as in `maybe-get-service`)